cadccutout/cadccutout/file_helpers/fits/fits_file_helper.py

```diff
@@ -162,43 +162,53 @@ def _pixel_cutout(self, hdu, cutout_dimension):
             data=hdu.data, cutout_dimension=cutout_dimension, wcs=wcs)
 
         self._post_sanitize_header(header, cutout_result)
-        self._write_cutout(header=header, data=cutout_result.data)
-
-    def _write_primary_hdu(self, hdu_list):
-        primary_hdu = hdu_list[0]
-        if isinstance(primary_hdu, PrimaryHDU):
-            logger.debug('Writing out Primary HDU.')
-            fits.append(
-                filename=self.output_writer, header=primary_hdu.header,
-                data=None, overwrite=False, output_verify='silentfix',
-                checksum='remove')
-            logger.debug('Wrote out Primary HDU.')
+        result = fits.ImageHDU(header=header, data=cutout_result.data)
+        self._fix_header(result.header, header)
+        return result
+
+    def _fix_header(self, target, original):
+        # overcome astropy's tendency to remove keywords from original headers
+        if 'BSCALE' in original:
+            target['BSCALE'] = original['BSCALE']
+        if 'BZERO' in original:
+            target['BZERO'] = original['BZERO']
+
+    def _add_primary_hdu(self, hdu_list, result_hdu_list):
+        if isinstance(hdu_list[0], PrimaryHDU):
```

Comment:
I suspect that the logic here should be different, see my code suggestion.
```suggestion
        if hdu_list[0].header.get("NAXIS", 0 ) == 0 :
```

Reply:
I guess you didn't want to have a value error raised when NAXIS is absent?  Not NAXIS means 'Not  a FITS file' so it would be awkward to have arrived this deep into the cutout code with an object that was not valid FITS.  No sure there is a need to change, just noting that this 'safe get' might be wrong as it's implying we can somehow work with hdu_list[0] not being a vaild FITS object.  This might be true... I'm not sure.